### PR TITLE
Fixed NULL check

### PR DESCRIPTION
**Pull request by rponnuru5**
Added a test case to check for NULL for rocrond_generate_long_long and rocrand_geenrate_char

### CODE REVIEW

File: test/test_rocrand_generate.cpp [hunk 1]
```diff
@@ -69,18 +69,23 @@ TEST_P(rocrand_generate_tests, int_test)
 
 TEST_P(rocrand_generate_tests, char_test)
 {
+    const size_t size = 12563;
+    unsigned char * data = NULL;
```
Comment (Naraenda) [inline test/test_rocrand_generate.cpp]:
```suggestion
    unsigned char * data = nullptr;
```

Idem for all other uses of `NULL`.

**Comment by Naraenda** [comment]
I meant replacing _all_ the `NULL` with `nullptr` in the proposed change, not just that single one 😅

**Comment by rponnuru5** [comment]
Hi Narendra, I have updated in all the places. pls review and approve the PR.

File: test/test_rocrand_generate.cpp [hunk 2]
```diff
@@ -145,10 +156,9 @@ TEST_P(rocrand_generate_tests, short_test)
 
 TEST(rocrand_generate_tests, neg_test)
```
Comment (Naraenda) [inline test/test_rocrand_generate.cpp:157]: After thinking on this a bit longer, why was it chosen to add the null check in the existing tests and not extend the "null" test into a [typed test suite](http://google.github.io/googletest/reference/testing.html#TYPED_TEST_SUITE)?

Comment (stanleytsang-amd) [reply test/test_rocrand_generate.cpp:157]: @rponnuru5 would you be able to refactor your changes to meet Nara's suggestion?

Comment (rponnuru5) [reply test/test_rocrand_generate.cpp:157]: @Naraenda , Pls provide more details about it.

Comment (Naraenda) [reply test/test_rocrand_generate.cpp:157]: I meant grouping the `nullptr` tests together with the already existing `nullptr` tests. Having the `nullptr` test in the generator parameterized tests doesn't make sense since we don't instantiate the different types of generators. There's no point in testing the same things multiple times with the exact same input parameters. I've implemented it here: https://github.com/ROCm/rocRAND/pull/519

**Comment by rponnuru5** [comment]
It is handled as part of 519 PR so closing this
